tests/test_algorithm.py

```diff
@@ -1735,25 +1736,24 @@ def test_close_position_future(self):
     def test_auto_close_future(self):
         metadata = {1: {'symbol': 'TEST',
                         'asset_type': 'future',
-                        'auto_close_date': self.days[3]}}
+                        'auto_close_date': self.env.trading_days[4]}}
         self.env.write_data(futures_data=metadata)
         algo = TestAlgorithm(sid=1, amount=1, order_count=1,
-                             instant_fill=True, commission=PerShare(0),
+                             commission=PerShare(0),
                              env=self.env)
         data = DataPanelSource(self.no_close_panel)
 
         # Check results
         results = algo.run(data)
 
-        expected_pnl = [0, 1, 2]
-        self.check_algo_pnl(results, expected_pnl)
-
-        expected_positions = [1, 1, 0]
+        expected_positions = [0, 1, 1, 0]
```

Comment:
Flipped the order of checking pnl vs. positions here since pnl is dependent on position values, so needed to get those passing first.